src/mscorlib/src/System/Span.cs

```diff
@@ -625,5 +636,266 @@ internal static unsafe void CopyTo<T>(ref T destination, ref T source, int eleme
                 }
             }
         }
+
+        internal static unsafe void ClearWithoutReferences(ref byte b, nuint byteLength)
+        {
+            if (byteLength == 0)
+                return;
+
+            fixed (byte* bytePointer = &b)
```

Comment:
How do you increment and cast ref byte to other types? Is it using Unsafe.Add and Unsafe.As<TFrom, TTo>APIs?

For example, if I am not pinning to a byte *, how would the syntax look for let's say:
```C#
*(int*)(bytePointer + 16) = (int)0;
// OR
if ((byteLength & 1) != 0)
```